File: 16-3sum-closest/3sum-closest.cs
public class Solution
{
    public int ThreeSumClosest(int[] nums, int target)
    {
        Array.Sort(nums);

        int closest = nums[0] + nums[1] + nums[2];

        for (int i = 0; i < nums.Length - 2; i++)
        {
            int left = i + 1;
            int right = nums.Length - 1;

            while (left < right)
            {
                int sum = nums[i] + nums[left] + nums[right];

                if (Math.Abs(target - sum) < Math.Abs(target - closest))
                {
                    closest = sum;
                }

                if (sum < target)
                {
                    left++;
                }
                else if (sum > target)
                {
                    right--;
                }
                else
                {
                    return sum; // Encontrou exatamente o target
                }
            }
        }

        return closest;
    }
}

File: 16-3sum-closest/3sum-closest.ts
function threeSumClosest(nums: number[], target: number): number {
    const duplicates = <T>(arr: T[]): boolean => {
        return new Set(arr).size !== arr.length;
    }

    const arr = nums.length;
    let closestSum = nums[0] + nums[1] + nums[2];

    for (let i = 0; i < arr; i++) {
        for(let j = i + 1; j < arr; j++){
            for (let k = j + 1; k < arr; k++){
                const countSum = nums[i] + nums[j] + nums[k];

                if(Math.abs(countSum - target) <
                    Math.abs(closestSum - target)){
                        closestSum = countSum
                    }
                
            }
        }
    }

    return closestSum;
}